src/OrchardCore/OrchardCore.Media.Core/DefaultMediaFileStore.cs

```diff
@@ -169,16 +169,24 @@ public virtual async Task<string> CreateFileFromStreamAsync(string path, Stream
 
                 foreach (var mediaCreatingEventHandler in _mediaCreatingEventHandlers)
                 {
-                    // Creating stream disposed by using.
-                    using var creatingStream = outputStream;
-
+                    var creatingStream = outputStream;
                     // Stop disposal of inputStream, as creating stream is the object to dispose.
                     inputStream = null;
 
                     // Outputstream must be created by event handler.
                     outputStream = null;
 
-                    outputStream = await mediaCreatingEventHandler.MediaCreatingAsync(context, creatingStream);
+                    try
+                    {
+                        outputStream = await mediaCreatingEventHandler.MediaCreatingAsync(context, creatingStream);
+                    }
+                    finally
+                    {
+                        if (creatingStream != outputStream && creatingStream != inputStream)
```

Comment:
As the `inputStream` is not getting disposed here anymore, we must modify the call site here:

https://github.com/OrchardCMS/OrchardCore/blob/e37292d1b53551c2e310d5fe7473c19f0a3be461/src/OrchardCore.Modules/OrchardCore.Media/Controllers/AdminController.cs#L258C31-L258C35

This must always reopen the resulting file.